projects/ngx-maplibre-gl/src/lib/source/canvas-source.component.ts

```diff
@@ -23,9 +24,9 @@ import { tap } from 'rxjs';
   changeDetection: ChangeDetectionStrategy.OnPush,
   hostDirectives: [{ directive: SourceDirective, inputs: ['id'] }],
 })
-export class CanvasSourceComponent implements OnChanges {
+export class CanvasSourceComponent extends Source implements OnChanges {
```

Comment:
We have decided to go with composition over inheritance for the sources...

Reply:
Should I just expect users to get a view child on the directive instead? Is it expected public api?

Reply:
I don't think so, no.
I guess you'll need to add the relevant methods to each source without inheritance...